plugin/keyring_vault/vault_curl.h

```diff
@@ -0,0 +1,52 @@
+#ifndef MYSQL_VAULT_CURL_H
+#define MYSQL_VAULT_CURL_H
+
+#include <my_global.h>
+#include <curl/curl.h>
```

Comment:
I don't understand why this curl.h through vault_curl.h dependency is bad, given that both header files are specifically for curl, that is, cohesion is high here. Compilation time argument does not work, curl.h is C. And IMHO pimpl does not qualify as "minimal effort". (forward declarations do, but they are not enough here)